lib/attrsets.nix

```
@@ -823,6 +823,57 @@ rec {
         ) a (attrNames n)
     ) {} list_of_attrs;
 
+  /**
+    Flatten an attribute set to the bottom name and value pairs as
+    one singular final attribute set, stripping everything else
+    above.
+
+    In a few words, this does what `lib.lists.flatten` does, but for
+    attribute sets, rather than lists.
+
+
+    # Inputs
+
+    `attrs`
+
+    : The top-level attribute set to flatten.
+
+    # Type
+
+    ```
+    flattenAttrs :: AttrSet -> AttrSet
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.attrsets.flattenAttrs` usage example
+
+    ```nix
+    flattenAttrs { a.b.c.d = "e"; one.two.three.four.five = 6; foo.bar = null; }
+    => { d = "e"; five = 6; bar = null; }
+    ```
+
+    :::
+  */
+  flattenAttrs = attrs:
+    let
+      op = acc: n:
+        let
+          v = attrs.${n};
+          x =
+            if (lib.isAttrs v)
```

Would be nice to make the predicate configurable

```suggestion
            if (pred v)
```

---

How would you recommend making this change or writing this? I have seen it around in some other functions, but am not too familiar. Sounds good to me.